### PR TITLE
Dart refactor go vaccine demo ruby vaccine demo

### PR DESCRIPTION
**Pull request by fundthmcalculus**
* Ignore any files changed under ruby/libs/, those are protoc generated files

### CODE REVIEW

**Review by sethjback — APPROVED** [review]

File: go/examples/vaccine_test.go [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+	template "github.com/trinsic-id/sdk/go/proto/verifiablecredentials/templates/v1"
+	credential "github.com/trinsic-id/sdk/go/proto/verifiablecredentials/v1"
+	"github.com/trinsic-id/sdk/go/services"
+	"testing"
+)
+
+func createTemplate(templateService services.CredentialTemplateService, t *testing.T) *template.TemplateData {
+	// createTemplate() {
+	templateRequest := &template.CreateCredentialTemplateRequest{Name: "VaccinationCertificate", AllowAdditionalFields: false, Fields: make(map[string]*template.TemplateField)}
+	templateRequest.Fields["firstName"] = &template.TemplateField{Description: "Given name"}
```
Comment (geel9) [inline go/examples/vaccine_test.go]:
```suggestion
	templateRequest.Fields["firstName"] = &template.TemplateField{Description: "First name of vaccine recipient"}
```

To ensure template parity across languages

File: docs/walkthroughs/vaccination.md [hunk 2]
```diff
@@ -76,10 +76,8 @@ This tutorial is meant to share the concepts of verifiable credentials, so feel
     In this project, we'll be following along the [`java/src/test/java/trinsic/VaccineDemo.java`](https://github.com/trinsic-id/sdk/blob/main/java/src/test/java/trinsic/VaccineDemo.java) in our [SDK repository](https://github.com/trinsic-id/sdk)
 
 
-    <!-- Ruby commented out because we don't have a Ruby sample for the rest of this yet
-    === "Ruby"
-        - [Install Instructions](../ruby/index.md) 
-    -->
+=== "Ruby"
```
Comment (geel9) [inline docs/walkthroughs/vaccination.md]: Should be moved to after `Go` to have parity with other code tabs on page

File: go/examples/vaccine_test.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+	templateRequest.Fields["lastName"] = &template.TemplateField{}
```
Comment (geel9) [inline go/examples/vaccine_test.go]:
```suggestion
	templateRequest.Fields["lastName"] = &template.TemplateField{Description: "Last name of vaccine recipient"}
```

File: go/examples/vaccine_test.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+	templateRequest.Fields["batchNumber"] = &template.TemplateField{Type: template.FieldType_NUMBER}
```
Comment (geel9) [inline go/examples/vaccine_test.go]:
```suggestion
	templateRequest.Fields["batchNumber"] = &template.TemplateField{Description: "Batch number of vaccine", Type: template.FieldType_NUMBER}
```

File: go/examples/vaccine_test.go [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+	templateRequest.Fields["countryOfVaccination"] = &template.TemplateField{}
```
Comment (geel9) [inline go/examples/vaccine_test.go]:
```suggestion
	templateRequest.Fields["countryOfVaccination"] = &template.TemplateField{Description: "Country in which the subject was vaccinated"}
```

File: go/services/service_base.go [hunk 6]
```diff
@@ -39,6 +39,8 @@ type Service interface {
 	// SetToken assigns the given auth token to the service. This token will be used for
 	// make all api calls
 	SetToken(token string)
+	// SetEcosystemId assigns the given ecosystem ID to the service.
+	SetEcosystemId(ecosystemId string)
```
Comment (geel9) [inline go/services/service_base.go:43]: Bless up 🙏

File: ruby/test/vaccine_demo.rb [hunk 7]
```diff
@@ -1,80 +1,131 @@
...
 require 'services/wallet_service'
 require 'json'
 
-def data_base_path
-  File.expand_path(File.join(File.dirname(__FILE__), "..", "..", "devops", "testdata"))
+def do_template(template_service)
+  # createTemplate() {
+  request = Trinsic::Template_V1::CreateCredentialTemplateRequest.new(name: 'VaccinationCertificate',
+                                                                      allow_additional_fields: false)
+  request.fields['firstName'] = Trinsic::Template_V1::TemplateField.new(description: 'Given name')
```
Comment (geel9) [inline ruby/test/vaccine_demo.rb]:
```suggestion
  request.fields['firstName'] = Trinsic::Template_V1::TemplateField.new(description: 'First name of vaccine recipient')
```

To ensure template parity across languages

File: ruby/test/vaccine_demo.rb [hunk 8]
```diff
@@ -1,80 +1,131 @@
...
+  request.fields['lastName'] = Trinsic::Template_V1::TemplateField.new
```
Comment (geel9) [inline ruby/test/vaccine_demo.rb]:
```suggestion
  request.fields['lastName'] = Trinsic::Template_V1::TemplateField.new(description: 'Last name of vaccine recipient')
```

File: ruby/test/vaccine_demo.rb [hunk 9]
```diff
@@ -1,80 +1,131 @@
...
+  request.fields['batchNumber'] = Trinsic::Template_V1::TemplateField.new(description: '')
```
Comment (geel9) [inline ruby/test/vaccine_demo.rb]:
```suggestion
  request.fields['batchNumber'] = Trinsic::Template_V1::TemplateField.new(description: 'Batch number of vaccine')
```

File: ruby/test/vaccine_demo.rb [hunk 10]
```diff
@@ -1,80 +1,131 @@
...
+  request.fields['countryOfVaccination'] = Trinsic::Template_V1::TemplateField.new(description: '')
```
Comment (geel9) [inline ruby/test/vaccine_demo.rb]:
```suggestion
  request.fields['countryOfVaccination'] = Trinsic::Template_V1::TemplateField.new(description: 'Country in which the subject was vaccinated')
```

File: go/examples/vaccine_test.go [hunk 11]
```diff
@@ -0,0 +1,142 @@
+package examples
```
Comment (geel9) [inline go/examples/vaccine_test.go:1]: Excellent, but it seems that `vaccination.md` isn't actually injecting from this file

**Review by geel9 — CHANGES_REQUESTED** [review]

File: go/examples/vaccine_test.go (hunk 4 above)
Comment (fundthmcalculus) [reply go/examples/vaccine_test.go]: I vote change to number, that was the original intent anyway.

Comment (geel9) [reply go/examples/vaccine_test.go]: > I vote change to number, that was the original intent anyway.

Roger that -- we'll leave it as a number here, and I'll track the changes to the other langs.

Comment (geel9) [reply go/examples/vaccine_test.go]: Updated the suggestion

**Review by geel9 — APPROVED** [review]

**Pull request merged**